Gems/Atom/Feature/Common/Code/Source/Decals/DecalTextureArray.cpp

```diff
@@ -122,42 +125,42 @@ namespace AZ
             return m_materials[index].m_materialAssetId;
         }
 
-        RHI::Size DecalTextureArray::GetImageDimensions() const
+        RHI::Size DecalTextureArray::GetImageDimensions(const DecalMapType mapType) const
         {
             AZ_Assert(m_materials.size() > 0, "GetImageDimensions() cannot be called until at least one material has been added");
             const int iter = m_materials.begin();
             // All textures in a texture array must have the same size, so just pick the first 
             const MaterialData& firstMaterial = m_materials[iter];
-            const auto& baseColorAsset = GetBaseColorImageAsset(firstMaterial.m_materialAssetData);
+            const auto& baseColorAsset = GetStreamingImageAsset(firstMaterial.m_materialAssetData, GetMapName(mapType));
             return baseColorAsset->GetImageDescriptor().m_size;
         }
 
-        const AZ::Data::Instance<AZ::RPI::StreamingImage>& DecalTextureArray::GetPackedTexture() const
+        const AZ::Data::Instance<AZ::RPI::StreamingImage>& DecalTextureArray::GetPackedTexture(const DecalMapType mapType) const
         {
-            return m_textureArrayPacked;
+            return m_textureArrayPacked[mapType];
         }
 
         bool DecalTextureArray::IsValidDecalMaterial(const AZ::RPI::MaterialAsset& materialAsset)
         {
-            return GetStreamingImageAsset(materialAsset, AZ::Name(BaseColorTextureMapName)).IsReady();
+            return GetStreamingImageAsset(materialAsset, GetMapName(DecalMapType_Diffuse)).IsReady();
         }
```

Comment:
What if I want a decal that only changes the normals and doesn't affect the color?

Reply:
As we discussed, if this is to be addressed in follow up work than I'm fine with this going in as is.